network/src/main/java/com/halcyonmobile/android/common/extensions/network/ContextExtension.kt

```diff
@@ -0,0 +1,45 @@
+package com.halcyonmobile.android.common.extensions.network
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+/**
+ * Checks if the device's active Network has internet capability.
+ *
+ * @return true if Device has Active Internet Available Network, false otherwise.
```

Comment:
Minor, thing but I would probably rephrase this to: `@return true if device has Available Network and Active Internet, false otherwise.` as it implies both the order and the requirements we're checking. How does that sound to you?